buildSrc/src/main/java/org/elasticsearch/gradle/PublishPlugin.java

```diff
@@ -0,0 +1,202 @@
+        // generatePomTask.configure(t -> t.dependsOn("generatePomFileForNebulaPublication"));
+
+        maybeConfigure(project.getTasks(), LifecycleBasePlugin.ASSEMBLE_TASK_NAME, assemble -> assemble.dependsOn(generatePomTask));
+
+        project.getTasks().withType(GenerateMavenPom.class).configureEach(pomTask -> pomTask.setDestination(new Callable<String>() {
+            @Override
+            public String call() throws Exception {
+                return String.format(
+                    "%s/distributions/%s-%s.pom",
+                    project.getBuildDir(),
+                    getArchivesBaseName(project),
+                    project.getVersion()
+                );
+            }
+        }));
+
+        PublishingExtension publishing = project.getExtensions().getByType(PublishingExtension.class);
+
+        project.getPluginManager().withPlugin("com.github.johnrengelman.shadow", plugin -> {
+            MavenPublication publication = publishing.getPublications().maybeCreate("shadow", MavenPublication.class);
+            ShadowExtension shadow = project.getExtensions().getByType(ShadowExtension.class);
+            shadow.component(publication);
+            // Workaround for https://github.com/johnrengelman/shadow/issues/334
+            // Here we manually add any project dependencies in the "shadow" configuration to our generated POM
+            publication.getPom().withXml(xml -> {
+                Node root = xml.asNode();
+                root.appendNode("name", project.getName());
+                root.appendNode("description", project.getDescription());
+                Node dependenciesNode = (Node) ((NodeList) root.get("dependencies")).get(0);
+                project.getConfigurations().getByName(ShadowBasePlugin.getCONFIGURATION_NAME()).getAllDependencies().all(dependency -> {
+                    if (dependency instanceof ProjectDependency) {
+                        Node dependencyNode = dependenciesNode.appendNode("dependency");
+                        dependencyNode.appendNode("groupId", dependency.getGroup());
+                        ProjectDependency projectDependency = (ProjectDependency) dependency;
+                        String artifactId = getArchivesBaseName(projectDependency.getDependencyProject());
+                        dependencyNode.appendNode("artifactId", artifactId);
+                        dependencyNode.appendNode("version", dependency.getVersion());
+                        dependencyNode.appendNode("scope", "compile");
+                    }
+                });
+            });
+        });
+
+        // Add git origin info to generated POM files
+        publishing.getPublications().withType(MavenPublication.class, publication -> {
+            publication.getPom().withXml(PublishPlugin::addScmInfo);
+
+            // have to defer this until archivesBaseName is set
+            project.afterEvaluate(p -> publication.setArtifactId(getArchivesBaseName(project)));
```

Comment:
I see we no longer do this explicitly after applying the shadow plugin. I put that there to workaround some very specific configuration ordering issues. Please confirm that the artifact id for the poms generated for the HLRC and JDBC drive are correct. They should be `elasticsearch-rest-high-level-client` and `x-pack-sql-jdbc`, respectively.

To be even more thorough its probably worth doing a diff on all POMs, given we fuck something up every time we touch publishing code. When I was testing this I ran `./gradlew generatePom` copied all the files to a folder, applied my changes, generated again, then did a diff to manually confirm my PR didn't change how POMs were generated.

Reply:
I double checked both those poms and they are correct. I will do a diff as you suggested.